editors/vscode/src/extension.ts

```diff
@@ -241,8 +249,84 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(
         vscode.commands.registerCommand("slint.newProject", newProject),
     );
+
+    startTelemetryTimer(context, telemetryLogger);
 }
 
 export function deactivate(): Thenable<void> | undefined {
     return common.deactivate();
 }
+
+function handleTelemetryEvent(
+    telemetryType: string,
+    globalState: vscode.Memento,
+) {
+    switch (telemetryType) {
+        case "preview_opened": {
+            const currentCount = globalState.get("preview_opened", 0);
+            globalState.update("preview_opened", currentCount + 1);
+            break;
+        }
+        case "property_changed": {
+            const currentCount = globalState.get("property_changed", 0);
+            globalState.update("property_changed", currentCount + 1);
+            break;
+        }
+        case "data_json_changed": {
+            const currentCount = globalState.get("data_json_changed", 0);
+            globalState.update("data_json_changed", currentCount + 1);
+            break;
+        }
+        default:
+            console.log("Received unknown telemetry event:", telemetryType);
+            break;
+    }
```

Comment:
How about this:

```suggestion
            const currentCount = globalState.get(telemetryType, 0);
            globalState.update(telemetryType, currentCount + 1);
```

That way we don't need to update this function if we add telemetry event.



Or another alternative:

```suggestion
            let telemetryState = globalState.get("telemetryState", {});
            telemetryState["telemetryType"] += 1;
            globalState.update(telemetryState, currentCount + 1);
```

And then you can send telemetryState to the server later.